src/Libraries/RevitNodes/Elements/Material.cs

```diff
@@ -232,11 +246,24 @@ public IEnumerable<Parameter> ThermalParameters
                         foreach (var parameter in thermal.Parameters)
                         {
                             Parameter p = new Parameter(parameter as Autodesk.Revit.DB.Parameter);
-                            if (!thermals.Contains(p)) thermals.Add(p);
+                            if (!thermals.Contains(p) && p.InternalParameter.Definition != null) thermals.Add(p);
```

Comment:
You can have a function to take `Parameters` as input, so you can reuse the code.